activitysim/benchmarking/latest.py

```diff
@@ -0,0 +1,181 @@
+            action="store_true",
+            help="""Skip running benchmarks that have previous successful
+            results""",
+        )
+        parser.add_argument(
+            "--skip-existing-failed",
+            action="store_true",
+            help="""Skip running benchmarks that have previous failed
+            results""",
+        )
+        parser.add_argument(
+            "--skip-existing-commits",
+            action="store_true",
+            help="""Skip running benchmarks for commits that have existing
+            results""",
+        )
+        parser.add_argument(
+            "--skip-existing",
+            "-k",
+            action="store_true",
+            help="""Skip running benchmarks that have previous successful
+            or failed results""",
+        )
+        parser.add_argument(
+            "--interleave-processes",
+            action="store_true",
+            default=False,
+            help="""Interleave benchmarks with multiple processes across
+            commits. This can avoid measurement biases from commit ordering,
+            can take longer.""",
+        )
+        parser.add_argument(
+            "--no-interleave-processes",
+            action="store_false",
+            dest="interleave_processes",
+        )
+        parser.add_argument(
+            "--no-pull", action="store_true", help="Do not pull the repository"
+        )
+
+        parser.set_defaults(func=cls.run_from_args)
+
+        return parser
+
+    @classmethod
+    def run_from_conf_args(cls, conf, args, **kwargs):
+        return cls.run(
+            conf=conf,
+            range_spec="HEAD^!",
+            steps=None,
+            bench=args.bench,
+            attribute=args.attribute,
+            parallel=args.parallel,
+            show_stderr=args.show_stderr,
+            quick=args.quick,
+            profile=args.profile,
+            env_spec=args.env_spec,
+            set_commit_hash=args.set_commit_hash,
+            dry_run=args.dry_run,
+            machine=args.machine,
+            skip_successful=args.skip_existing_successful or args.skip_existing,
+            skip_failed=args.skip_existing_failed or args.skip_existing,
+            skip_existing_commits=args.skip_existing_commits,
+            record_samples=True,
+            append_samples=True,
+            pull=not args.no_pull,
+            interleave_processes=args.interleave_processes,
+            launch_method=args.launch_method,
+            **kwargs
+        )
+
+
+class Batch(Run):
+    @classmethod
+    def setup_arguments(cls, subparsers):
+        parser = subparsers.add_parser(
+            "batch",
+            help="Run a set of benchmark suites based on a batch file. "
+            "Simply give the file name, which should be a text file "
+            "containing a number of activitysim benchmark commands.",
+            description="Run a set of benchmark suites based on a batch file.",
+        )
```

Comment:
This is great.  I think this functionality should be included in the main documentation file so people will know about it.

Reply:
Done.